pkg/login/social/connectors/social_base.go

```diff
@@ -209,3 +210,14 @@ func getRoleFromSearch(role string) (org.RoleType, bool) {
 
 	return org.RoleType(cases.Title(language.Und).String(role)), false
 }
+
+func validateInfo(info *social.OAuthInfo) error {
+	if info.ClientId == "" {
+		return ssosettings.ErrEmptyClientId.Errorf("clientId is empty")
+	}
+	if info.ClientSecret == "" {
```

Comment:
Client Secret can be empty in Authorization Code+PKCE flow. 

We have customers that depend on that behavior and it's currently our recommended for Okta

Reply:
Thanks for the info, I'll remove the validation of the client secret for now.